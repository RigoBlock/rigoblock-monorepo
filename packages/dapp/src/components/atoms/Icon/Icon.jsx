import './Icon.scss'
import PropTypes from 'prop-types'
import React from 'react'
import Tooltip from '../Tooltip'
import classNames from 'classnames'

export const ICON_SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large'
}

const Icon = ({ size, type, onClick, tooltipText, className }) => {
  const classProps = classNames('material-icons', 'icon', size, className)
  return tooltipText ? (
    <div className={classProps} onClick={onClick}>
      <span className="tooltip-hover">{type}</span>
      <Tooltip tooltipText={tooltipText} />
    </div>
  ) : (
    <div className={classProps} onClick={onClick}>
      {type}
    </div>
  )
}

Icon.propTypes = {
  type: PropTypes.string.isRequired,
  tooltipText: PropTypes.string,
  onClick: PropTypes.func,
  className: PropTypes.string,
  size: PropTypes.oneOf([ICON_SIZES.SMALL, ICON_SIZES.MEDIUM, ICON_SIZES.LARGE])
}

Icon.defaultProps = {
  size: ICON_SIZES.MEDIUM,
  onClick: () => {},
  className: '',
  tooltipText: ''
}

export default Icon
