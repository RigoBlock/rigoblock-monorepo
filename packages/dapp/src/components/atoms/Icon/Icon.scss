@import "../../_settings/variables.scss";

.icon {
  color: $silver-sand;
  display: inline;
  position: relative;
  cursor: pointer;

  &.small {
    font-size: 10px;
  }

  &.medium {
    font-size: $large-font-size;
  }

  &.large {
    font-size: $xl-font-size;
  }
}

.tooltip:hover {
  &:before {
    font-family: $font-family;
    font-size: $medium-font-size;
    color: $black;
    background: $secondary-bg-color;
    border-radius: $border-radius;
    line-height: $line-height;
    box-shadow: $box-shadow;
    bottom: calc(100% + #{$dist-s});
    white-space: nowrap;
    content: attr(tooltip);
    padding: $dist-s;
    left: 50%;
    transform: translateX(-50%);
    position: absolute;
  }

  &:after {
    bottom: calc(100% + #{$dist-xs});
    content:"";
    background: $secondary-bg-color;
    width: $dist-s;
    height: $dist-s;
    box-shadow: 1px 1px 0.009em 0 rgba(0, 0, 0, 0.05);
    left: 50%;
    transform: translateX(-50%) rotate(45deg);
    position: absolute;
  }
}
