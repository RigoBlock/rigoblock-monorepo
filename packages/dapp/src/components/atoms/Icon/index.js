export { default, ICON_SIZES } from './Icon'
