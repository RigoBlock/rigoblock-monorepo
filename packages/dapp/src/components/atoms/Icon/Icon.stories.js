import '../../_settings/_base.scss'
import { select, text, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import Icon, { ICON_SIZES } from './Icon'
import React from 'react'

storiesOf('Atoms/Icon', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div
      style={{
        marginTop: '100px',
        marginLeft: '100px'
      }}
    >
      {story()}
    </div>
  ))
  .add('default', () => (
    <Icon
      size={select('Icon size', ICON_SIZES, ICON_SIZES.SMALL)}
      type={text('Icon type', 'help')}
    >
      Small Icon
    </Icon>
  ))
  .add('with text', () => (
    <Icon
      size={select('Icon size', ICON_SIZES, ICON_SIZES.SMALL)}
      type={text('Icon type', 'help')}
      tooltipText={text('Icon text', 'I am a tooltip!')}
    >
      Small Icon
    </Icon>
  ))
