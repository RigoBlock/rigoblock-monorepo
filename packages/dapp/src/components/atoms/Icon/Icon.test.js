import { shallow } from 'enzyme'
import Icon from './Icon.jsx'
import toJson from 'enzyme-to-json'

const props = {
  type: 'help',
  onClick: jest.fn()
}

describe('Icon component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Icon, props)))
    ).toMatchSnapshot()
  })
  it('calls onClick function when clicked', () => {
    const tooltip = shallow(createComponentWithProps(Icon, props))
    tooltip.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })
})
