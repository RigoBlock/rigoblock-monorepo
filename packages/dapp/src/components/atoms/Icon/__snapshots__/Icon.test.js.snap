// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
<div
  className="material-icons icon medium"
  onClick={[MockFunction]}
>
  help
</div>
`;
